src/pages/WalletsItemBackup/WalletsItemBackupView.js

```diff
@@ -0,0 +1,265 @@
+|}
+
+export const STEPS: WalletsItemBackupSteps = {
+  PASSWORD: 'PASSWORD',
+  BACKUP_FORM: 'BACKUP_FORM',
+}
+
+const WALLETS_BACKUP_INITIAL_VALUES: FormFields = {
+  password: '',
+}
+
+export class WalletsItemBackupView extends Component<Props, StateProps> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      name: '',
+      data: '',
+      passphrase: '',
+      derivationPath: '',
+      currentStep: STEPS.PASSWORD,
+      customType: null,
+    }
+  }
+
+  getTitle = (): string => {
+    switch (this.state.currentStep) {
+      case STEPS.PASSWORD:
+        return t`Enter Security Password`
+
+      case STEPS.BACKUP_FORM:
+        return t`Backup wallet`
+
+      default:
+        return ''
+    }
+  }
+
+  handleSubmit = async (values: FormFields): Promise<?FormFields> => {
+    const {
+      goBackToWallets,
+      walletId,
+      internalKey,
+      salt,
+    }: Props = this.props
+
+    switch (this.state.currentStep) {
+      case STEPS.PASSWORD: {
+        const {
+          encrypted,
+          name,
+          customType,
+          derivationPath,
+        }: Wallet = walletsPlugin.getWallet(walletId)
+
+        const derivedKey: Uint8Array = await deriveKeyFromPassword(
+          values.password || '',
+          salt,
+        )
+
+        try {
+          const internalKeyDec: Uint8Array = decryptInternalKey(
+            internalKey,
+            derivedKey,
+          )
+
+          this.setState({
+            name,
+            customType,
+          })
+
+          switch (customType) {
+            case 'privateKey': {
+              if (!encrypted.privateKey) {
+                throw new WalletInconsistentDataError()
+              }
+
+              const data: string = decryptData({
+                key: internalKeyDec,
+                data: encrypted.privateKey,
+              })
+
+              this.setState({ data })
+
+              break
+            }
+
+            case 'xprv': {
+              if (!encrypted.xprv) {
+                throw new WalletInconsistentDataError()
+              }
+
+              const data: string = decryptData({
+                key: internalKeyDec,
+                data: encrypted.xprv,
+              })
+
+              this.setState({ data })
+
+              break
+            }
+
+            case 'mnemonic': {
+              if (!encrypted.mnemonic) {
+                throw new WalletInconsistentDataError()
+              }
+
+              const data: string = decryptData({
+                key: internalKeyDec,
+                data: encrypted.mnemonic,
+              })
+
+              if (!encrypted.passphrase) {
+                throw new WalletInconsistentDataError()
+              }
+
+              const passphrase: string = decryptData({
+                key: internalKeyDec,
+                data: encrypted.passphrase,
+              })
+
+              this.setState({ data })
+
+              this.setState({
```

Comment:
It's work as designed: two asynchronous `setState` calls one by one?

Reply:
it will be fixed